### PR TITLE
AJ-1674: handle logical decimals in Parquet files

### PR DESCRIPTION
**Pull request by davidangb**
[AJ-1674](https://broadworkbench.atlassian.net/browse/AJ-1674)

WDS's conversion of certain Parquet data types was lacking. Specifically, this is a problem with Parquet files exported from BigQuery (i.e. handled by cWDS as part of importing GCS-based TDR snapshots) when the BigQuery tables contain `NUMERIC` columns. 

`NUMERIC` columns are exported to Parquet as:
```
physical_type: FIXED_LEN_BYTE_ARRAY
logical_type: Decimal(precision=38, scale=9)
converted_type (legacy): DECIMAL
```
this empirical finding is confirmed by [Google's documentation](https://cloud.google.com/bigquery/docs/reference/storage#arrow_schema_details). Note that the precision and scale may differ, but it will always be a _physical_ FIXED_LEN_BYTE_ARRAY and a _logical_ Decimal.

WDS's code in `AvroRecordConverter` did not handle these logical types. It simply called `.toString()` on the FIXED_LEN_BYTE_ARRAY, which resulted in incorrect values.

Without the changes in this PR, the new unit test would fail with:
```
java.lang.AssertionError: 
Expecting actual:
  ["[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]",
    "[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 13, 28, -17, 0]",
    "[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -104, -106, -128]"]
to contain exactly in any order:
  [0, 0.01, 0.22]
elements not found:
  [0, 0.01, 0.22]
and elements not expected:
  ["[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]",
    "[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 13, 28, -17, 0]",
    "[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -104, -106, -128]"]
```
showing that the value we created for `Fixed` logical decimals was pretty far off.

Finally, note that [AJ-1674](https://broadworkbench.atlassian.net/browse/AJ-1674) actually concerns behavior in Import Service. This PR fixes the same issue in WDS; I will handle the Import Service issue separately.

[AJ-1674]: https://broadworkbench.atlassian.net/browse/AJ-1674?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AJ-1674]: https://broadworkbench.atlassian.net/browse/AJ-1674?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: service/src/main/java/org/databiosphere/workspacedataservice/dataimport/AvroRecordConverter.java [hunk 1]
```diff
@@ -185,6 +188,40 @@ public Object convertAttributeType(Object attribute) {
...
+
+      // now, try to turn the BigDecimal back into an int or a double; this truncates the
+      // precision to the actual value. Example: 0.220000000 will become 0.22
+      Object returnValue;
+      try {
+        returnValue = bigDecimal.intValueExact();
+      } catch (ArithmeticException ae) {
+        returnValue = bigDecimal.doubleValue();
+      }
+      return returnValue;
```
Comment (nawatts) [inline service/src/main/java/org/databiosphere/workspacedataservice/dataimport/AvroRecordConverter.java:216]: Up to you, but slightly more concise as...?

```suggestion
      try {
        return bigDecimal.intValueExact();
      } catch (ArithmeticException ae) {
        return bigDecimal.doubleValue();
      }
```

**Review by nawatts — APPROVED** [review]

File: service/src/test/java/org/databiosphere/workspacedataservice/dataimport/tdr/ParquetDataTypesTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    ParquetRecordConverter converter = new ParquetRecordConverter(testTable, mapper);
+
+    InputFile numericsFile =
+        HadoopInputFile.fromPath(
+            new Path(numericsParquet.getURL().toString()), new Configuration());
+
+    // ACT - read via ParquetReader and convert via ParquetRecordConverter
+    try (ParquetReader<GenericRecord> avroParquetReader =
+        TdrManifestQuartzJob.readerForFile(numericsFile)) {
```
Comment (jladieu) [inline service/src/test/java/org/databiosphere/workspacedataservice/dataimport/tdr/ParquetDataTypesTest.java:55]: optional supernit: static import to get this try onto one line?

File: service/src/test/java/org/databiosphere/workspacedataservice/dataimport/tdr/ParquetDataTypesTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      assertThat(records).hasSize(3);
+
+      // Each column in numeric.parquet has the same values, but
+      // each column has a different schema. The [0, 0.01, 0.22] values are the expected
+      // values in numerics.parquet.
+      List.of("numeric_default", "numeric_custom", "bignumeric_default", "bignumeric_custom")
+          .forEach(
+              attributeName ->
+                  assertThat(getColumnValues(records, attributeName))
+                      .describedAs("for column %s", attributeName)
```
Comment (jladieu) [inline service/src/test/java/org/databiosphere/workspacedataservice/dataimport/tdr/ParquetDataTypesTest.java:87]: cool use of the DSL!

**Review by jladieu — APPROVED** [review]

File: service/src/main/java/org/databiosphere/workspacedataservice/dataimport/AvroRecordConverter.java (hunk 1 above)
Comment (davidangb) [reply service/src/main/java/org/databiosphere/workspacedataservice/dataimport/AvroRecordConverter.java]: ah, good call, my code was kinda sloppy! fixed.

**Pull request merged**